libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -75,85 +94,248 @@ public NavigationMapboxMap(MapView mapView, MapboxMap mapboxMap) {
     initializeCamera(mapboxMap);
   }
 
+  /**
+   * Adds a marker icon on the map at the given position.
+   * <p>
+   * The icon used for this method can be defined in your theme with
+   * the attribute navigationViewDestinationMarker.
+   *
+   * @param context  to retrieve the icon drawable from the theme
+   * @param position the point at which the marker will be placed
+   */
   public void addMarker(Context context, Point position) {
     Marker marker = createMarkerFromIcon(context, position);
     mapMarkers.add(marker);
   }
 
+  /**
+   * Clears all markers on the map that have been added by this class.
+   * <p>
+   * This will not clear all markers from the map entirely.  Does nothing
+   * if no markers have been added.
+   */
   public void clearMarkers() {
     removeAllMarkers();
   }
 
+  /**
+   * Updates the location icon on the map and way name data (if found)
+   * for the given {@link Location}.
+   *
+   * @param location to update the icon and query the map
+   */
   public void updateLocation(Location location) {
     locationLayer.forceLocationUpdate(location);
     updateMapWaynameWithLocation(location);
   }
 
+  /**
+   * Can be used to automatically drive the map camera / route updates and arrow
+   * once navigation has started.
+   * <p>
+   * These will automatically be removed in {@link MapboxNavigation#onDestroy()}.
+   *
+   * @param navigation to add the progress listeners
+   */
   public void addProgressChangeListener(MapboxNavigation navigation) {
     mapRoute.addProgressChangeListener(navigation);
     mapCamera.addProgressChangeListener(navigation);
   }
 
+  /**
+   * Can be used to store the current state of the map in
+   * {@link android.support.v4.app.FragmentActivity#onSaveInstanceState(Bundle, PersistableBundle)}.
+   * <p>
+   * This method uses {@link NavigationMapboxMapInstanceState}, stored with the
+   * {@link NavigationMapboxMap#MAP_INSTANCE_STATE_KEY}.
+   *
+   * @param outState to store state variables
+   */
+  public void onSaveInstanceState(Bundle outState) {
```

Comment:
Really like how we took advantage of the naming in `restoreFrom` to make the API easier to read and understand ❤️
Have we considered doing something similar here?